stackrox/stackrox
# ROX-13519: API Token expiration notification

## Description

API tokens are generated with a hardcoded expiration of 1 year.

In order to help administrators keep track of the tokens about to expire, a new background task checking for tokens about to expire is being added to central.
The first iteration will start with a look checking every day for items about to expire and will issue audit traces for tokens that will expire within a week.

## Checklist
~~- [ ] Determined and documented upgrade steps~~
~~- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~~

If any of these don't apply, please comment below.

## Testing Performed

CI run for added tests.

Manual testing:
- alter in the codebase the thresholds to have tokens with 1 hour liftetime, that trigger notifications every 6 minutes for the last 24 minutes of their lifetime.
- deploy central.
- create tokens sets at different points in time
- check that notifications that the tokens are about to expire get sent according to the constraints.
